Dynatrace/dynatrace-operator
# Revert GORM changes

## Description

[K8S-10709](https://dt-rnd.atlassian.net/browse/K8S-10709)

Reverts/removes all the changes that introduced GORM and the new database schema, as it causes more trouble then its worth.

The state should be similar to what we had in 1.1, with a few additions:
- Digest is no longer used for the image-installer (so when CodeModulesImage is set)
  - Using the "old" database schema, we store the `ImageUri` (base64 encoded, see comment in code) in the `ImageDigest` column
    - Smallest and least intrusive change possible, as it's a bug fix
- Added extra safety checks in the GC, when handling the codemodule binary folders.
  - This is necessary so if someone moves from the "new"(GORM) database schema to the "old" one, will not have problems.
    - Without the checks, we might try to remove the binary folders that are still mounted as we don't have any information about them.


## How can this be tested?
Try scenarios where you upgrade from a released 1.2 version, while codemodules are mounted, and oneagents are rolled out.

Example: (what I tried)
1. Deploy Operator 1.2.1
2. Deploy cloudnative DynaKube (with a given codeModulesImage set)
3. Deploy Sample apps
4. Update cloudnative DynaKube to not have codeModulesImage
5. Upgrade the Operator to branch version.
6. Check that the GC doesn't delete codeModulesImage binary as the Sample apps use it
7. Delete Sample apps (they don't get stuck)
8. Delete DynaKube (oneagents don't get stuck)
